builder/build.go

```diff
@@ -15,13 +15,22 @@ import (
 	v1execute "github.com/alexellis/go-execute/pkg/v1"
 	"github.com/openfaas/faas-cli/schema"
 	"github.com/openfaas/faas-cli/stack"
+	"github.com/openfaas/faas-cli/utils"
 	vcs "github.com/openfaas/faas-cli/versioncontrol"
 )
 
 // AdditionalPackageBuildArg holds the special build-arg keyname for use with build-opts.
 // Can also be passed as a build arg hence needs to be accessed from commands
 const AdditionalPackageBuildArg = "ADDITIONAL_PACKAGE"
 
+func init() {
+	// Check the ENV VAR "CI" to know if faas-cli is being invoked in a CI/CD env
+	// and change the default dir location & permissions for files to be accessible by all users
+	if utils.IsRunningInCI() {
```

Comment:
I am not sure i am a fan of adding stuff to the `init`. As a general rule this should be avoided. I would rather see `permissions` passed as an argument to the method that needs it.  Not only is it easier to test but it is also easier to read/debug if there is an error.

Reply:
@LucasRoesler makes sense for testability. I'll make the relevant changes.

Reply:
@LucasRoesler I've tried not to touch the existing vars as they're declared at the top. I don't know if my current commit makes a lot of sense for what you would like to see, but considering I don't want to change the location of these global vars to a function, the current code made sense to me.
I'm thinking about not to affect additional code that could be added in the future to these files, changing the way these vars are accessed or set. What do you think?